### PR TITLE
Image upload

### PR DESCRIPTION
**Pull request by evabrinkmann**

### CODE REVIEW

**Comment by arisslin** [comment]
I get this error after 'npm start' in the console: FirebaseError: projectId must be a string in FirebaseApp.options

I think it is somethin about the firebase settings to which i have no access to, so i will only review the code itself without testing the ui in the browser.

File: src/components/Bookmark.js [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
     event.stopPropagation()
     onBookmarkClick(id)
   }
 
   return (
-    <BookmarkStyled
-      onClick={event => handleBookmarkClick(event)}
-      active={bookmarkStatus}
-    ></BookmarkStyled>
+    <>
```
Comment (arisslin) [inline src/components/Bookmark.js]: You don't need the '<>' because your code is already wrapped into Tooltip.

File: src/components/Bookmark.js [hunk 2]
```diff
@@ -1,17 +1,27 @@
 import React from 'react'
 import styled from 'styled-components'
+import Tooltip from '@material-ui/core/Tooltip'
 
-export default function Bookmark({ onBookmarkClick, id, bookmarkStatus }) {
+export default function Bookmark({
```
Comment (arisslin) [inline src/components/Bookmark.js]: Think about PropTypes.

File: src/components/Card.js [hunk 3]
```diff
@@ -9,8 +9,6 @@ import { useToggle } from 'react-hooks-lib'
 
 Card.propTypes = {
```
Comment (arisslin) [inline src/components/Card.js:10]: Think about PropTypes for the other props of Card

File: src/components/CardHead.js [hunk 4]
```diff
@@ -19,24 +28,74 @@ export default function CardHead({
   handleDelete,
   id,
   pathname,
+  setProfiles,
+  profiles,
 }) {
   const { on, toggle } = useToggle(false)
```
Comment (arisslin) [inline src/components/CardHead.js:31]: What is 'on' and what is 'toggle'. Maybee you can specify this more precise in the naming.

File: src/components/CardHead.js [hunk 5]
```diff
@@ -19,24 +28,74 @@ export default function CardHead({
...
-      {on === false
-        ? pathname === '/profile-pool' && (
-            <DeleteSign onClick={() => handleDelete(id)}>
-              <img src={deleteIcon} alt="trashbox" />
-            </DeleteSign>
-          )
-        : ''}
-      <img src={img} alt="portrait" />
+      {on === false && pathname === '/profile-pool' && (
+        <DeleteButton onClick={() => handleDelete(id)}>
```
Comment (arisslin) [inline src/components/CardHead.js]: I would create for DeleteButton an own file where you handle the component. Then your code in this file is a bit shorter and cleaner.

File: src/components/CardHead.js [hunk 6]
```diff
@@ -19,24 +28,74 @@ export default function CardHead({
...
+          <Tooltip title="Delete card" placement="left">
+            <img src={deleteIcon} alt="trashbox" />
+          </Tooltip>
+        </DeleteButton>
+      )}
+      <UploadWrapper>
+        <img src={img} alt="portrait" />
+        {pathname === '/profile-pool' && (
+          <UploadButton htmlFor="imageInput">
```
Comment (arisslin) [inline src/components/CardHead.js]: Like the DeleteButton I would create an own file where you handle the component. Then your code in this file is a bit shorter and cleaner. Maybe you can make a common component named Button and can specify it's role over props. They both have similar intern components like Tooltip. So this is a good reason to think about.

File: src/components/CardHead.js [hunk 7]
```diff
@@ -19,24 +28,74 @@ export default function CardHead({
...
+            </Tooltip>
+          </UploadButton>
+        )}
+      </UploadWrapper>
       <h1>{name}</h1>
       <h2>{title}</h2>
     </StyledHead>
   )
+
+  function updateProfileImage(url) {
```
Comment (arisslin) [inline src/components/CardHead.js]: Nice!

File: src/components/CardHead.js [hunk 8]
```diff
@@ -19,24 +28,74 @@ export default function CardHead({
...
+    const updatedProfile = { ...profiles[index], imgUrl: url }
+    setProfiles([
+      ...profiles.slice(0, index),
+      updatedProfile,
+      ...profiles.slice(index + 1),
+    ])
+    saveProfilesToLocal('profiles', profiles)
+  }
+
+  function handleUpload(event) {
```
Comment (arisslin) [inline src/components/CardHead.js]: Rename in `handleProfileImageUpload()`

File: src/components/CardHead.js [hunk 9]
```diff
@@ -19,24 +28,74 @@ export default function CardHead({
...
+    const image = event.target.files[0]
+    const uploadTask = storage.ref(`user-images/${image.name}`).put(image)
+    uploadTask.on(
```
Comment (arisslin) [inline src/components/CardHead.js]: Make a function which handles uploadTask.on()

```
function uploadProfileImage(task, imageName) {
  task.on(
      'state_changed',
      snapshot => {},
      error => {
        console.error(error)
      },
      () => {
        storage
          .ref('user-images')
          .child(imageName)
          .getDownloadURL()
          .then(url => {
            updateProfileImage(url)
          })
      }
    )
}
```

File: src/components/CardList.js [hunk 10]
```diff
@@ -12,6 +12,7 @@ export default function CardList({
   onDelete,
```
Comment (arisslin) [inline src/components/CardList.js:16]: Think about proptypes for the other props.

File: src/pages/PoolPage.js [hunk 11]
```diff
@@ -4,7 +4,12 @@ import Header from '../components/Header'
 import { useLocation } from 'react-router-dom'
 import styled from 'styled-components/macro'
 
-export default function PoolPage({ profiles, onDelete, onBookmarkClick }) {
+export default function PoolPage({
```
Comment (arisslin) [inline src/pages/PoolPage.js:15]: Think about PropTypes.

**Review by arisslin — CHANGES_REQUESTED** [review]
Think about PropTypes and do a little bit refactor (mostly in CardHead) then it should be done. The rest looks fine. Good work!

File: src/components/Bookmark.js (hunk 1 above)
Comment (evabrinkmann) [reply src/components/Bookmark.js]: True, thank  you. Forgot to delete it.

File: src/components/CardHead.js (hunk 4 above)
Comment (evabrinkmann) [reply src/components/CardHead.js:31]: Hmm not sure how to specify it as it toggles on click and if I name it after my components thats more confusing.

File: src/components/CardHead.js (hunk 5 above)
Comment (evabrinkmann) [reply src/components/CardHead.js]: That was my idea in the first place but thats easier said than done. I tried it already and couldn't manage it. My app crashed. Piers finally told me not to do it. The problem is that the delete component needs to be in the component cardhead as when the card is toggled the delete button is not shown. If you have an idea how to manage it feel free to let me know : )

File: src/components/CardHead.js (hunk 6 above)
Comment (evabrinkmann) [reply src/components/CardHead.js]: I tried here as well to create an own component but the css of the delete function and the upload function are intertwined. I consulted Piers he said to let go of it. If you feel brave enough we can have a look together.

File: src/components/CardHead.js (hunk 7 above)
Comment (evabrinkmann) [reply src/components/CardHead.js]: I had help : )

File: src/components/CardHead.js (hunk 9 above)
Comment (evabrinkmann) [reply src/components/CardHead.js]: not sure to fully understand, maybe we can talk about it this afternoon?

File: src/pages/PoolPage.js (hunk 11 above)
Comment (evabrinkmann) [reply src/pages/PoolPage.js:15]: I will go through all my components now

**Review by arisslin — APPROVED** [review]

**Pull request merged**